src/detect-dce-stub-data.c

```diff
@@ -91,20 +91,9 @@ static InspectionBuffer *GetDCEData(DetectEngineThreadCtx *det_ctx,
     InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
     if (buffer->inspect == NULL) {
         uint32_t data_len = 0;
-        uint8_t *data = NULL;
-
-        DCERPCState *dcerpc_state = txv;
-        if (dcerpc_state == NULL)
-            return NULL;
+        const uint8_t *data = NULL;
 
-        if (flow_flags & STREAM_TOSERVER) {
-            data_len = dcerpc_state->dcerpc.dcerpcrequest.stub_data_buffer_len;
-            data = dcerpc_state->dcerpc.dcerpcrequest.stub_data_buffer;
-        } else if (flow_flags & STREAM_TOCLIENT) {
-            data_len = dcerpc_state->dcerpc.dcerpcresponse.stub_data_buffer_len;
-            data = dcerpc_state->dcerpc.dcerpcresponse.stub_data_buffer;
-        }
-        if (dcerpc_state->dcerpc.dcerpchdr.packed_drep[0] & 0x10) {
+        if (rs_dcerpc_set_buffer(txv, &data, &data_len, flow_flags)) {
```

Comment:
this function appears to be able to not touch `data` and `data_len`, so we need a check for when `data == NULL || data_len == 0` before calling `InspectionBufferSetup`